patterns/legacy/exim

```diff
@@ -1,13 +1,16 @@
 EXIM_MSGID [0-9A-Za-z]{6}-[0-9A-Za-z]{6}-[0-9A-Za-z]{2}
 EXIM_FLAGS (<=|[-=>*]>|[*]{2}|==)
 EXIM_DATE %{YEAR:exim_year}-%{MONTHNUM:exim_month}-%{MONTHDAY:exim_day} %{TIME:exim_time}
-EXIM_PID \[%{POSINT}\]
+EXIM_PID \[%{POSINT:pid}\]
 EXIM_QT ((\d+y)?(\d+w)?(\d+d)?(\d+h)?(\d+m)?(\d+s)?)
 EXIM_EXCLUDE_TERMS (Message is frozen|(Start|End) queue run| Warning: | retry time not reached | no (IP address|host name) found for (IP address|host) | unexpected disconnection while reading SMTP command | no immediate delivery: |another process is handling this message)
-EXIM_REMOTE_HOST (H=(%{NOTSPACE:remote_hostname} )?(\(%{NOTSPACE:remote_heloname}\) )?\[%{IP:remote_host}\])
+EXIM_REMOTE_HOST (H=(%{NOTSPACE:remote_hostname} )?(\(%{NOTSPACE:remote_heloname}\) )?\[%{IP:remote_host}\])(?::%{POSINT:remote_port})?
 EXIM_INTERFACE (I=\[%{IP:exim_interface}\](:%{NUMBER:exim_interface_port}))
 EXIM_PROTOCOL (P=%{NOTSPACE:protocol})
 EXIM_MSG_SIZE (S=%{NUMBER:exim_msg_size})
 EXIM_HEADER_ID (id=%{NOTSPACE:exim_header_id})
 EXIM_SUBJECT (T=%{QS:exim_subject})
 
+EXIM_MESSAGE_ARRIVAL %{EXIM_DATE:timestamp} (?:%{EXIM_PID} )?%{EXIM_MSGID:exim_msgid} (?<exim_flags><=) (?:%{EXIM_FLAGS:exim_flags} )?(?<exim_status>[a-z:] )?%{EMAILADDRESS:exim_sender_email}(?: (?:%{EXIM_REMOTE_HOST}|%{EXIM_INTERFACE}|%{EXIM_PROTOCOL}|%{EXIM_MSG_SIZE}|%{EXIM_HEADER_ID}|%{EXIM_SUBJECT}|(?:[A-Za-z]{1,4}=%{NOTSPACE})))*(?: for %{EMAILADDRESS:exim_recipient_email})?
```

Comment:
Separately, I'm not _quite_ sure I understand this sequence:

~~~
(?<exim_flags><=) (?:%{EXIM_FLAGS:exim_flags} )?
~~~

To me that reads: _required `<=` sequence_ followed by an optional `EXIM_FLAGS`. This is odd because `EXIM_FLAGS` includes the `<=` message arrival flag, so there _should_ be no other flags? The spec does not seem to indicate that there will be one-and-only-one flag per line, but all of its examples exhibit this behaviour. If multiple flags _are_ expected, can we guarantee that the _first_ flag is this message arrival sequence? Can we guarantee that there will be at most two?

Reply:
it's been a while but it seems like I intentionally left it in - if so totally my bad I did not include a spec.
in some cases there might be multiple flags in a single log entry but I am not finding any examples where that would hold (for message arrival) atm. believe we can remove that and start with just matching one flag `<=` here, for now.

Reply:
removed the optional `(?:%{EXIM_FLAGS:exim_flags} )?`